sympy/functions/elementary/trigonometric.py

```diff
@@ -2221,6 +2221,8 @@ class asec(InverseTrigonometricFunction):
 
     @classmethod
     def eval(cls, arg):
+        if arg.is_zero:
+            return S.NaN
```

Comment:
Perhaps better reference could have been [this](http://functions.wolfram.com/ElementaryFunctions/ArcSec/03/01/).